Improve Asset loading Warning Issue #482

bernardthered opened the pull request.

Commit:
Proudly adding myself to the contributors file

Commit:
Improve warning msg when asset's file is missing

The new sentence was copied from the issue: https://github.com/ppb/pursuedpybear/issues/482

Commit:
remove email from contributors table

AstraLuma reviewed (changes requested):
If I'm reading this correctly, this will add this warning text for every kind of asset with a `file_missing` behavior, but the text is only applicable to `ppb.Image`.

Commit:
Only show image-specific message when asset is an image

Move the image-specific warning message to a new class attribute `not_found_message` on the Image class, and also define that class attribute (and a comment explaining it) on the AbstractAsset ABC.

bernardthered commented:
@AstraLuma good catch! I corrected that issue, let me know what you think of the solution, I'd be happy to change anything about that :-)

bernardthered commented:
@pathunstrom it looks like another spurious unit test error (this time the error is "❗ Agent is not responding!"). Would you be able to press the rerun link when you get a chance?

Commit:
Merge branch 'canon' into improve-asset-loading-warning-issue-482

AstraLuma commented on ppb/assetlib.py:
```diff
@@ -165,6 +165,10 @@ class AbstractAsset(abc.ABC):
     real/file assets.
     """
+    # Can be overridden on child classes to add an additional message to show after "File not
+    # found: %r.". See Image.not_found_message for an example.
```
This should be #: so it gets picked up by sphinx

bernardthered replied:
I didn't know about that #: syntax, thanks for the pointer. I just read a bit about it and updated those two comment lines.

AstraLuma reviewed (changes requested):
Minor changes. Otherwise, I'm happy with this design and implementation.

Commit:
Update comment on not_found_message to be sphinx-friendly

Also update the warning message to be more explicit.

bernardthered commented:
I also updated the warning message to be more explicit, let me know if you think the original is better and I can change it back.

AstraLuma reviewed (approved):
bors r+
